### PR TITLE
Add (and style) the basic Catalog page layout

### PR DESCRIPTION
**Pull request by hojberg**
## Problem
We want to render a catalog page (https://github.com/unisonweb/codebase-ui/issues/144)

## Solution
CleanShot 2021-12-14 at 16 06 39@2x

(this screenshot is the current look, but needs more work in terms of hero background and input focus)

* To support the Catalog page, add a unison-share css folder for the catalog page and a new Card module for cards holding each category in  the catalog.
* Rename `UI.Page` to `UI.PageLayout` to better communicate what that module is for.
* Add a way to generate a Route from a Project (exclusively to the Unison Share target) to allow clicking to through to the project page from the catalog

## Caveats/Notes
There's a bunch of hardcoded mock data that'll be removed when we load this from the server for real (empty categories are not expected to show up on the final page).

### CODE REVIEW

File: src/UnisonShare/Page/Catalog.elm [hunk 1]
```diff
@@ -1,26 +1,33 @@
 module UnisonShare.Page.Catalog exposing (..)
 
-import Api
-import Dict exposing (Dict)
+-- import Api
```
Comment (hojberg) [inline src/UnisonShare/Page/Catalog.elm:3]: Disabled to satisfy linter - will make it back as we start to load real data.

File: src/UnisonShare/Page/Catalog.elm [hunk 2]
```diff
@@ -34,14 +41,27 @@ type alias Model =
...
-                |> Api.perform env.apiBasePath
+        categories =
+            Set.empty
+
+        model =
+            Success
+                { catalog = Catalog categories
+                , query = ""
+                }
     in
```
Comment (hojberg) [inline src/UnisonShare/Page/Catalog.elm:63]: Mocking a `Success` respinse.

File: src/Project.elm [hunk 3]
```diff
@@ -10,13 +9,18 @@ type Owner
 
 
 type alias Project a =
-    { a | owner : Owner, name : FQN, hash : Hash }
+    { a | owner : Owner, name : FQN }
```
Comment (hojberg) [inline src/Project.elm:12]: Remove the Hash field for now to make it easier to mock (creating a `Hash` returns a `Maybe`)—it'll be back in a later iteration.

**Review by pchiusano — APPROVED** [review]
Sweet!!!

**Pull request merged**